### PR TITLE
feat: typings and small changes in exports

### PR DESCRIPTION
**Pull request by BrooklynKing**

### CODE REVIEW

File: src/core/default-modules.ts [hunk 1]
```diff
@@ -1,45 +1,85 @@
...
+  IPlayer as IPreviewServicePlayer,
+} from '../modules/ui/preview-service/preview-service';
 import PreviewThumbnail from '../modules/ui/preview-thumbnail/preview-thumbnail';
 import PreviewFullSize from '../modules/ui/preview-full-size/preview-full-size';
 
 import { TooltipService } from '../modules/ui/core/tooltip';
 
 const { asClass } = DependencyContainer;
 
+export type IPlayer = IRootContainerPlayer &
```
Comment (maxvektor) [inline src/core/default-modules.ts]: Not sure, you should name type starting with `I`.

File: src/core/default-modules.ts [hunk 2]
```diff
@@ -1,45 +1,85 @@
...
 import TimeControl from '../modules/ui/controls/time/time';
 import VolumeControl from '../modules/ui/controls/volume/volume';
 import FullScreenControl from '../modules/ui/controls/full-screen/full-screen';
-import Logo from '../modules/ui/controls/logo/logo';
-import DownloadButton from '../modules/ui/controls/download/download';
-
-import PreviewService from '../modules/ui/preview-service/preview-service';
+import Logo, { IPlayer as ILogoPlayer } from '../modules/ui/controls/logo/logo';
+import DownloadButton, {
+  IPlayer as IDownloadButtonPlayer,
```
Comment (maxvektor) [inline src/core/default-modules.ts]: Maybe we should rename all this `IPlayer`s to more specific name?

File: src/core/default-modules.ts (hunk 1 above)
Comment (bodia-uz) [reply src/core/default-modules.ts]: We have a convention. All types should start from `I` (declared by `type` and `interface).

File: src/constants/errors.ts [hunk 3]
```diff
@@ -1,4 +1,4 @@
-const Errors = {
+const Error = {
```
Comment (bodia-uz) [inline src/constants/errors.ts]: mb if you rename it to pascal case, use `enum`?

File: src/modules/event-emitter/event-emitter.ts [hunk 4]
```diff
@@ -18,29 +18,28 @@ const isPromiseAvailable = (function() {
...
 
-export default class EventEmitterModule extends EventEmitter
-  implements IEventEmitter {
+class EventEmitterModule extends EventEmitter implements IEventEmitter {
   static moduleName = 'eventEmitter';
 
   /**
    * Method for adding listeners of events inside player.
-   * You can check all events inside `Playable.UI_EVENTS` and `Playable.VIDEO_EVENTS`
+   * You can check all events inside `Playable.UIEvent` and `Playable.VideoEvent`
```
Comment (bodia-uz) [inline src/modules/event-emitter/event-emitter.ts]: not sure if this `JSDoc` also should be updated. You exported it `UPPERCASES`

File: src/modules/playback-engine/live-state-engine.ts [hunk 5]
```diff
@@ -1,17 +1,12 @@
...
+import { EngineState, VideoEvent, LiveState, UIEvent } from '../../constants';
 
 import { IEventEmitter, IEventMap } from '../event-emitter/types';
 import { ILiveStateEngineDependencies, IPlaybackEngine } from './types';
 
-const SEEK_BY_UI_EVENTS = [
-  UI_EVENTS.GO_FORWARD_WITH_KEYBOARD,
-  UI_EVENTS.GO_BACKWARD_WITH_KEYBOARD,
-  UI_EVENTS.PROGRESS_CHANGE,
+const SEEK_BY_UIEvent = [
```
Comment (bodia-uz) [inline src/modules/playback-engine/live-state-engine.ts]: Is it `UPPER` or `Pascal` case?;)

File: src/modules/playback-engine/state-engine.spec.ts [hunk 6]
```diff
@@ -2,8 +2,8 @@ import { expect } from 'chai';
 import * as sinon from 'sinon';
 import { EventEmitter } from 'eventemitter3';
 
-import { VIDEO_EVENTS, EngineState } from '../../constants';
-import StateEngine, { NATIVE_VIDEO_EVENTS_TO_STATE } from './state-engine';
+import { VideoEvent, EngineState } from '../../constants';
+import StateEngine, { NATIVE_VideoEvent_TO_STATE } from './state-engine';
```
Comment (bodia-uz) [inline src/modules/playback-engine/state-engine.spec.ts]: cat

**Pull request merged**